src/mapnode.cpp

```diff
@@ -64,8 +64,12 @@ u8 MapNode::getFaceDir(const NodeDefManager *nodemgr,
 			f.param_type_2 == CPT2_COLORED_4DIR)
 		return getParam2() & 0x03;
 	if (allow_wallmounted && (f.param_type_2 == CPT2_WALLMOUNTED ||
-			f.param_type_2 == CPT2_COLORED_WALLMOUNTED))
-		return wallmounted_to_facedir[getParam2() & 0x07];
+			f.param_type_2 == CPT2_COLORED_WALLMOUNTED)) {
+		u8 wmountface = getParam2() & 0x07;
+		if (wmountface >= DWM_END)
+			wmountface = 0;
```

Comment:
you could use `wmountface = MYMIN(wmountface, DWM_END);` to implement this shorter